tests/phpt/cl/191_toString_magic_method_string_builder.php

```diff
@@ -0,0 +1,14 @@
+@ok
+<?php
+
+class A {
+  function __toString() {
+    return "hello";
+  }
+}
+
+function demo(A $a) {
+  echo "$a and $a";
```

Comment:
Let's extend the testcase with the following:
```
// class A
public function getCopy() : A { return $this; }
public function getArray: array { return [$this]; }

echo "{$a->getCopy()} {$a->getArray()[0]}";
```
(we want to test, that arbitrary expressions work inside string builds)

Reply:
Added.